diff --git a/consumers.cabal b/consumers.cabal
index f5e5aea..e1a335e 100644
--- a/consumers.cabal
+++ b/consumers.cabal
@@ -31,7 +31,7 @@ library
                     Database.PostgreSQL.Consumers.Components,
                     Database.PostgreSQL.Consumers.Utils
 
-  build-depends:    base              >= 4.9    && < 4.13
+  build-depends:    base              >= 4.9    && < 4.14
                   , containers        >= 0.5    && < 0.7
                   , exceptions        >= 0.10   && < 0.11
                   , extra             >= 1.6    && < 1.7
@@ -43,7 +43,7 @@ library
                   , monad-time        >= 0.3    && < 0.4
                   , mtl               >= 2.2    && < 2.3
                   , stm               >= 2.4    && < 2.6
-                  , time              >= 1.6    && < 1.9
+                  , time              >= 1.6    && < 2.0
                   , transformers-base >= 0.4    && < 0.5
 
   hs-source-dirs:   src
