diff --git a/happstack-server.cabal b/happstack-server.cabal
index e516232..23fd5e3 100644
--- a/happstack-server.cabal
+++ b/happstack-server.cabal
@@ -70,7 +70,7 @@ Library
                        Paths_happstack_server
 
   if flag(network-uri)
-     build-depends:    network     >  2.6 && < 2.8,
+     build-depends:    network     >  2.6,
                        network-uri >= 2.6 && < 2.7
   else
      build-depends:    network               < 2.6
