diff --git a/hslogger.cabal b/hslogger.cabal
index 83b2601..0a91be6 100644
--- a/hslogger.cabal
+++ b/hslogger.cabal
@@ -70,12 +70,8 @@ library
       , deepseq    >= 1.1 && < 1.5
       , time       >= 1.2 && < 1.10
       , old-locale >= 1.0 && < 1.1
-
-    if flag(network--GT-3_0_0)
-      build-depends: network-bsd >= 2.8.1 && <2.9,
-                     network >= 3.0 && <3.2
-    else
-      build-depends: network >= 2.6 && <2.9
+      , network-bsd >= 2.8.1 && <2.9
+      , network >= 3.0 && <3.2
 
     if !os(windows)
       Build-Depends: unix >= 2.4.2 && < 2.8
