diff --git a/src/Mixpanel/Engage.hs b/src/Mixpanel/Engage.hs
index f18f141..492a1f8 100644
--- a/src/Mixpanel/Engage.hs
+++ b/src/Mixpanel/Engage.hs
@@ -44,10 +44,10 @@ set token distinctid properties = do
   let obj = runJSONGen $ do
         J.value "$token" token
         J.value "$distinct_id" distinctid
-        -- $ip must be set either to 0 or the IP supplied, otherwise Mixpanel
+        -- ip must be set either to 0 or the IP supplied, otherwise Mixpanel
         -- thinks they live in Dublin, Ireland (AWS EU server location)
         jvalue $ fromMaybe (IP "0") (find isip properties)
-        -- $ip must be in outer json
+        -- ip must be in outer json
         J.object "$set" $ forM_ (filter (not . isip) properties) jvalue
       jsstring = J.encode obj
       jsb64 = B.toString $ B64.encode $ B.fromString jsstring
@@ -74,7 +74,7 @@ add token distinctid properties = do
   let obj = runJSONGen $ do
         J.value "$token" token
         J.value "$distinct_id" distinctid
-        -- $ip must be in outer json
+        -- ip must be in outer json
         maybe (return ()) jvalue $ find isip properties
         J.object "$add" $ forM_ (filter (not . isip) properties) jvalue
       jsstring = J.encode obj
